x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/edit_analytics_flyout.tsx

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiForm,
+  EuiFormRow,
+  EuiOverlayMask,
+  EuiSelect,
+  EuiTitle,
+} from '@elastic/eui';
+
+import { useMlKibana } from '../../../../../contexts/kibana';
+import { ml } from '../../../../../services/ml_api_service';
+import {
+  memoryInputValidator,
+  MemoryInputValidatorResult,
+} from '../../../../../../../common/util/validators';
+import { DataFrameAnalyticsListRow, DATA_FRAME_TASK_STATE } from './common';
+import {
+  useRefreshAnalyticsList,
+  UpdateDataFrameAnalyticsConfig,
+} from '../../../../common/analytics';
+
+interface EditAnalyticsJobFlyoutProps {
+  closeFlyout: () => void;
+  item: DataFrameAnalyticsListRow;
+}
+
+let mmLValidator: (value: any) => MemoryInputValidatorResult;
+
+export const EditAnalyticsFlyout: FC<EditAnalyticsJobFlyoutProps> = ({ closeFlyout, item }) => {
+  const [allowLazyStart, setAllowLazyStart] = useState<string>('');
+  const [description, setDescription] = useState<string>('');
+  const [modelMemoryLimit, setModelMemoryLimit] = useState<string>('');
+  const [mmlValidationError, setMmlValidationError] = useState<string | undefined>();
+
+  const {
+    services: { notifications },
+  } = useMlKibana();
+  const { refresh } = useRefreshAnalyticsList();
+
+  const { id: jobId, config } = item;
+  const { state } = item.stats;
+
+  const updateButtonDisabled =
+    allowLazyStart === '' && description === '' && modelMemoryLimit === '';
+
+  useEffect(() => {
+    if (mmLValidator === undefined) {
+      mmLValidator = memoryInputValidator();
+    }
+    // validate mml and create validation message
+    if (modelMemoryLimit !== '') {
+      const validationResult = mmLValidator(modelMemoryLimit);
+      if (validationResult !== null && validationResult.invalidUnits) {
```

Comment:
Can the Update button be disabled if there is invalid input in the model memory limit input:

![image](https://user-images.githubusercontent.com/7405507/86384921-5885ec00-bc87-11ea-9ce9-e07d7fd83a5c.png)

Reply:
Added in 1086c4f031b0e8ffdc7801b0665d57fbf6ea53c6